### PR TITLE
Adding UnevaluatedExpr() Tutorial

### LINKED ISSUE #12230: Write documentation and examples to UnevaluatedExpr
The class `sympy.core.expr.UnevaluatedExpr` can potentially be very useful.

- Write some documentation as to its usages in the tutorials and/or introduction to SymPy.
- Especially important, mark the difference between `UnevaluatedExpr` and `evaluate=False`.
- Show that SymPy printers seamlessly support `UnevaluatedExpr` (and report a bug if any issues arise).

Related on StackOverflow: http://stackoverflow.com/questions/42486283/preventing-a-multiplication-expression-evaluating-in-sympy/42503593#42503593

### PR DESCRIPTION
**Pull request by mohit3011**
This Pull Request Fixes #12230

### CODE REVIEW

**Comment by mohit3011** [comment]
@Upabjojr , please review the changes . Thanks

**Comment by Upabjojr** [comment]
I would put the code here: 
https://github.com/sympy/sympy/blob/master/doc/src/tutorial/manipulation.rst

(Advanced expression manipulation).

**Comment by Upabjojr** [comment]
Also link the class here: https://github.com/sympy/sympy/blob/master/doc/src/modules/core.rst#expr-1

**Comment by mohit3011** [comment]
@Upabjojr  , I have changed it according to you, please review it . Thanks

**Comment by Upabjojr** [comment]
Strange... Travis did not start.

File: doc/src/tutorial/manipulation.rst [hunk 1]
```diff
@@ -483,3 +483,49 @@ traversals easy.  We could have also written our algorithm as
   have ``-x`` or ``1/x``.  It is also done for speed efficiency because
   singletonized objects can be compared by ``is``.  The unique objects for
   each singletonized class can be accessed from the ``S`` object.
+
+``UnevaluatedExpr``
+===================
```
Comment (Upabjojr) [inline doc/src/tutorial/manipulation.rst]: The tutorial should not be about the class `UnevaluatedExpr`, rather about generic methods to keep expressions unevaluated. I think we should also mention `evaluate=False`.

**Comment by Upabjojr** [comment]
Added some fixes. Tests are probably still failing.

**Comment by mohit3011** [comment]
@Upabjojr , do you know why the tests are failing ?

**Comment by Upabjojr** [comment]
> do you know why the tests are failing ?

Your example outputs are wrong.

**Comment by mohit3011** [comment]
I checked it on terminal in sympy , it can't be wrong

**Comment by Yathartha22** [comment]
-  line 520 the result should be `2*x` since the `expr` variable has not been updated it still has a value of `x + x`.
-  line 526 the result should be `(5/7)*(3/4)`
- line 531 the result should be `x*1/x`
-  line 539 should be `2*x`

**Comment by Upabjojr** [comment]
> I checked it on terminal in sympy , it can't be wrong

Maybe you use a different printer.

**Comment by mohit3011** [comment]
@Upabjojr  , I checked all the examples once again . There doesn't seem any error in those . I would like to point to core.rst , because I am not really sure about that.
One more thing Since the examples are on continuous basis , so are we assumming that once we import certain class from sympy then we can use it in all example or we have to import it every time.

**Comment by Upabjojr** [comment]
https://travis-ci.org/sympy/sympy/jobs/206607119

There is still a minor problem. If tests fail, just click on the X near the commit hashtag to see why.

> once we import certain class from sympy then we can use it in all example or we have to import it every time.

The doctests are generally sequential while executing the code, so different blocks are treated sequentially in every test. I think it's on a class-level or RST module-level (that is, higher scope is not sequential anymore), but I'm not so sure.

**Comment by mohit3011** [comment]
@Upabjojr , I don't know but the travis test is not opening up . This problem has been persistent since last night.

**Comment by Upabjojr** [comment]
> I don't know why but the travis test is not opening up . This problem has been persistent since last night.

Yeah, I think there's some problems with Travis today. I was able to access it one hour ago.

Otherwise, run the tests locally.

**Comment by mohit3011** [comment]
@Upabjojr , now I think its fine. One runtime error is due to some other module, I guess.

**Comment by Upabjojr** [comment]
> Warning, treated as error:
> /home/travis/build/sympy/sympy/doc/src/tutorial/manipulation.rst:471: WARNING: Title underline too short.

That's our file. We need a shorter title.

**Comment by mohit3011** [comment]
@Upabjojr  , finally :)

**Comment by mohit3011** [comment]
@Upabjojr , how can I run doc tests locally , because I can only see an error related to job time limit exceeded on Travis . Thanks

**Comment by Upabjojr** [comment]
> how can I run doc tests locally , because I can only see an error related to job time limit exceeded on Travis . Thanks

I think that this time Travis had a spurious failure. Restarted.

There is a directory `bin` in SymPy with scripts to run the tests.

**Comment by mohit3011** [comment]
@Upabjojr , I have run tests for other modules from bin by using the command like `python bin/test "module_name"` for testing "module_name".py but for running doc test what should I write ?

**Comment by Upabjojr** [comment]
> I have run tests for other modules from bin by using the command like python bin/test "module_name" for testing "module_name".py but for running doc test what should I write ?

One of those scripts calls the command, just look at the code.

**Pull request merged**